texastribune/donations
# Replace ds-toolbox tasks with queso-tools

#### What's this PR do?

Updates our front-end task runners to pull from [`queso-tools`](https://www.npmjs.com/package/@texastribune/queso-tools)

#### Why are we doing this? How does it help us?

It's a grand plan to simplify our design system iteration process. Splitting our design system repo into 2 repos helps us directly include our asset library (`queso-ui`) in /texastribune and include the task runner helpers (`queso-tools`) in tt-docker-base.

#### How should this be manually tested?
Build all the things
`make`
`make restart` - this will run `yarn run dev`

Ensure the following:
1. No build errors outputted in your console
2. `static/build/styles.json` and `static/build/*.css` are present
3. `templates/includes/base_icons.html` is still intact and contains symbol elements
4. Visually, the site remains unchanged.


#### How should this change be communicated to end users?
N/A

#### Are there any smells or added technical debt to note?
Yes, dart sass feels slower, despite [what they say](https://github.com/sass/dart-sass#why-dart). Under the hood of `queso-tools`, I changed the sass compiler `sass` instead of `node-sass`. If that starts to get in our way, it's a simple change back.

#### What are the relevant tickets?

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***


#### queso roll-out plan

* [x] Convert _ds-toolbox_ compiling tasks to queso-tools
* [x] **Convert _donations_ compiling tasks to queso-tools** <= we are here
* [ ] Deprecate styles.js and icons.js and related dependencies in _ds-toolbox_
* [ ] Restructure _ds-toolbox_ to form queso-ui package (leaving assets folder intact)
* [ ] Convert _donations_ asset grab to queso-ui
* [ ] Convert _texastribune_ asset grab to queso-ui
